File: CinemaApp/src/app/app-logic/timeslot.ts
import { Time } from '@angular/common';
import { ScreeningRoomData } from './screening-room-data';
import { Film } from './film/film';
import { Format } from './format/format';

export class Timeslot {
  id!: number;
  startTime!: Time;
  endTime!: Time;
  screeningRoom?: ScreeningRoomData;
  format?: Format;
  movie?: Film;

  constructor(item?: Partial<Timeslot>) {
    Object.assign(this, item);
  }
}

File: CinemaAPI/CinemaAPI/Models/TimeSlot.cs
namespace CinemaAPI.Models
{
    public class TimeSlot
    {
        public int Id { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public float Price {get; set; }
        public ScreeningRoom ScreeningRoom { get; set; }
        public Format Format { get; set; }
        public Film Film { get; set; }
    }
}
